higher_health/forms.py

```diff
@@ -347,13 +343,70 @@ def medical_fields(self):
 
 
 class HealthCheckLogin(forms.Form):
-    phone = forms.CharField(
-        widget=TextInput(attrs={"placeholder": "Phone number"}),
+    msisdn = forms.CharField(
+        label="Enter your mobile number",
+        widget=TextInput(attrs={"placeholder": "Mobile number"}),
         required=True,
-        max_length=100,
+        validators=[za_phone_number],
     )
-    fullname = forms.CharField(
-        widget=TextInput(attrs={"placeholder": "Last name"}),
+
+    def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop("request")
+        super().__init__(*args, **kwargs)
+
+    def clean_msisdn(self):
+        number = phonenumbers.parse(self.cleaned_data["msisdn"], "ZA")
+        return phonenumbers.format_number(number, phonenumbers.PhoneNumberFormat.E164)
+
+    def send_otp_sms(self, msisdn):
+        otp = "".join(secrets.choice(string.digits) for _ in range(6))
+        h = hmac.new(settings.SECRET_KEY.encode(), otp.encode(), digestmod=sha256)
+        otp_hash = base64.b64encode(h.digest()).decode()
+        self.request.session["otp_hash"] = otp_hash
+
+        if rapidpro:
+            try:
+                rapidpro.create_flow_start(
+                    params={"otp": otp},
+                    flow=settings.RAPIDPRO_SEND_OTP_SMS_FLOW,
+                    urns=[f"tel:{msisdn}"],
+                )
+            except TembaException:
+                self.add_error(
+                    "msisdn",
+                    "We're unable to send you an OTP at this time. Please try again.",
+                )
+
+    def clean(self):
+        cleaned_data = super().clean()
+        msisdn = cleaned_data.get("msisdn")
+        if msisdn:
+            self.send_otp_sms(msisdn)
+
+
+class HealthCheckOTP(forms.Form):
+    otp = forms.CharField(
+        label="Enter 6 digit pin sent via SMS",
+        widget=NumberInput(attrs={"placeholder": "One Time Pin"}),
         required=True,
-        max_length=100,
+        min_length=6,
+        max_length=6,
     )
+
+    def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop("request")
+        super().__init__(*args, **kwargs)
+
+    def verify_otp(self, otp):
+        try:
+            session_otp = self.request.session.pop("otp_hash")
```

Comment:
I think this will mean that if they ever type the OPT incorrectly, then if they type it correctly, they won't be able to be let through.

I think we should only remove the `otp_hash` key from the session if the comparison is successful.